src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
Bad luck. 
```
assert((uintmax_t)uc->uc_argt <= LONG_MAX);
```
still throws the "always true" warning on 64bits.
I seem not to be able to find an expression that doesn't generate a warning in both 32 and 64bits.
Only solution I have by now is using preprocessor:
```
#if SIZEOF_LONG <= 4
    assert(uc->uc_argt <= LONG_MAX);
#endif
```

I haven't needed that until now. Can you think of other option?